### ISSUE TITLE
Two or one digit number format of time results in different time zones

### ISSUE THREAD

**Issue by andbag** [zopefoundation/DateTime#5]
Hi,
we observed an odd behaviour of time string parsing

Example:
```python
    from DateTime import DateTime
    t1 = DateTime('2016-04-21 13:05')
    t2 = DateTime('2016-04-21 13:5')
    assert (t1 == t2), '%s is not equal to %s' % (t1, t2)
```
results in
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AssertionError: 2016/04/21 13:05:00 GMT+0 is not equal to 2016/04/21 13:05:00 GMT+2

In summary, one digit number for hour, minute or second always returns TZ GMT+0. Two digit number format of time (hh:mm:ss) returns local TZ. Is this a bug or feature?

**Comment by tseaver** [comment]
I would call it a bug that the one-digit string doesn't raise an exception:  it isn't a valid format.  The behavior you are seeing is `DateTime` trying too hard to parse the string:  when it doesn't match ISO 8601 format, it falls back to more ad hoc parsing, and assumes local time.

Because of backward-compatibilty, we won't change `DateTime` to raise exceptions for strings like the one in your second example.
